web/app/scripts/src/directives/proportion-field.es.js

```diff
@@ -46,13 +46,17 @@ module.directive('proportionField', [
                     ProportionField.addErrors(scope.fields, errors || {});
                 });
 
-                scope.$watch('model', (values) => {
-                    if (values === null) {
+                scope.$watch('model', (newModel, oldModel) => {
+                    if (!newModel || !oldModel) {
                         scope.model = {};
+                        return;
                     }
-                    scope.total = ProportionField.calculateTotal(values);
-                    if (values) {
-                        ProportionField.addPercentages(scope.fields, values, scope.total);
+
+                    const newSum = ProportionField.calculateTotal(newModel)
+                    const changedIndex = ProportionField.getChangedFieldIndex(newModel, oldModel);
```

Comment:
This only needs to be calculated if `newSum > 100)` so could be within the `if`.